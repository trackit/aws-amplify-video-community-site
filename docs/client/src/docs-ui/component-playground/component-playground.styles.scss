amplify-authenticator {
  --container-height: auto;
  --container-display: inline;
}
