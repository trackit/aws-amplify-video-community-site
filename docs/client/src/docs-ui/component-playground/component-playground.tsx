import {Component, h, Prop} from "@stencil/core";
import {AuthenticatorWithSlots} from "./custom-components/authenticator-with-slots";
import {CustomComponentName} from "./component-playground.types";

@Component({
  tag: "docs-component-playground",
  styleUrl: "component-playground.styles.scss",
})
export class DocsComponentPlayground {
  /** Name of component used in the playground */
  @Prop() readonly componentName: CustomComponentName;

  renderCustomComponent(componentName: CustomComponentName) {
    switch (componentName) {
      case "AuthenticatorWithSlots":
        return AuthenticatorWithSlots();
    }
  }

  render() {
    return this.renderCustomComponent(this.componentName);
  }
}
