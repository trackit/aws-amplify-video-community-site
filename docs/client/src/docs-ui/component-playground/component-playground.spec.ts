import {DocsComponentPlayground} from "./component-playground";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-component-playground", () => {
  describe("Component logic", () => {
    let componentPlayground: DocsComponentPlayground;
    beforeEach(() => (componentPlayground = new DocsComponentPlayground()));
  });

  describe("Render logic", () => {
    it("should render empty", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsComponentPlayground],
            html: `<docs-component-playground />`,
          })
        ).root,
      ).toMatchSnapshot();
    });

    it("should render with AuthenticatorWithSlots custom component", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsComponentPlayground],
            html: `<docs-component-playground component-name="AuthenticatorWithSlot"></docs-component-playground>`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
