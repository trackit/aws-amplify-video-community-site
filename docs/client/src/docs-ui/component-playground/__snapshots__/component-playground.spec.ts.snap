// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-component-playground Render logic should render empty 1`] = `<docs-component-playground></docs-component-playground>`;

exports[`docs-component-playground Render logic should render with AuthenticatorWithSlots custom component 1`] = `<docs-component-playground component-name="AuthenticatorWithSlot"></docs-component-playground>`;
